File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/masterdata/core/implementation/mappers/MasterDataServiceMapper.java
package io.zenwave360.example.clinicaltool.modules.masterdata.core.implementation.mappers;

import io.zenwave360.example.clinicaltool.common.mappers.BaseMapper;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.inbound.dtos.*;

import org.mapstruct.AfterMapping;
import java.util.List;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;
import org.mapstruct.factory.Mappers;
import org.springframework.data.domain.Page;

@Mapper(uses = { BaseMapper.class })
public interface MasterDataServiceMapper {

    MasterDataServiceMapper INSTANCE = Mappers.getMapper(MasterDataServiceMapper.class);

// input mappings
    // MasterDataFilternull-MasterData listMasterDataOfType
        MasterData asMasterData(MasterDataType type, String lang);
    @Mapping(target = "id", ignore = true)MasterData update(@MappingTarget MasterData entity, MasterDataType type, String lang);
    // MasterData-MasterData updateMasterData
        MasterData asMasterData(MasterData input);
    @Mapping(target = "id", ignore = true)MasterData update(@MappingTarget MasterData entity, MasterData input);
// output mappings
    // MasterData-MasterDataKeyValue listMasterDataOfType
    MasterDataKeyValue asMasterDataKeyValue(MasterData entity);
        List<MasterDataKeyValue> asMasterDataKeyValueList(List<MasterData> entity);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/masterdata/mappers/MasterDataServiceMapper.kt
package io.zenwave360.example.clinicaltool.modules.masterdata.mappers

import io.zenwave360.example.clinicaltool.common.mappers.BaseMapper
import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*
import io.zenwave360.example.clinicaltool.modules.masterdata.dtos.*
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingTarget
import org.mapstruct.factory.Mappers

@Mapper(uses = [BaseMapper::class])
interface MasterDataServiceMapper {

    companion object {
        val INSTANCE: MasterDataServiceMapper = Mappers.getMapper(MasterDataServiceMapper::class.java)
    }

    // input mappings
    // MasterDataFilternull-MasterData listMasterDataOfType
    fun asMasterData(type: MasterDataType?, lang: String?): MasterData {
        return MasterData().apply {
            // TODO: implement this method
            // this.type = type
            // this.lang = lang
        }
    }

    fun update(entity: MasterData, type: MasterDataType?, lang: String?): MasterData {
        return entity.apply {
            // TODO: implement this method
            // this.type = type
            // this.lang = lang
        }
    }

    // MasterData-MasterData updateMasterData

    fun asMasterData(input: MasterData): MasterData

    @Mapping(target = "id", ignore = true) fun update(@MappingTarget entity: MasterData, input: MasterData): MasterData

    // output mappings
    // MasterData-MasterDataKeyValue listMasterDataOfType
    fun asMasterDataKeyValue(entity: MasterData): MasterDataKeyValue

    fun asMasterDataKeyValueList(entity: List<MasterData>): List<MasterDataKeyValue>
}
